src/ActionList/Item.tsx

```diff
@@ -310,10 +307,12 @@ export function Item(itemProps: Partial<ItemProps> & {item?: ItemInput}): JSX.El
     onKeyPress,
     children,
     onClick,
-    id,
+    id: _id,
     ...props
   } = itemProps
 
+  const id = useMemo(() => _id ?? uniqueId(), [_id])
```

Comment:
I'm not sure we can rely on `id` being unique here.  For example, if I have a list of cats and a list of dogs, they could have overlapping _data_ ids that are not unique at a global level, but are unique within the list.  We should probably just do `uniqueId` in all cases.

Reply:
Good point! I mistakenly thought `id` was being applied as the `id` attribute, which is supposed to be globally-unique, as opposed to the `data-id`. Revised in https://github.com/primer/components/pull/1320/commits/254a4a3a7092f00b1eafce450a508b6b758006d8